### ISSUE TITLE
seqkit stats restarts processing after x number of files

### ISSUE THREAD

**Issue by neutronstar21** [shenwei356/seqkit#467]
Hi, here are the preliminaries:

- [ ] Make sure you are using the latest version by `seqkit version -u`.
seqkit v2.8.2

### Describe your issue in detail

- [ ] Please copy and paste the command you ran and the error information if reported.
seqkit stats ./raw_unclean/lane_1/*.FASTQ.gz
or
seqkit stats ./raw_unclean/lane_1/*.FASTQ.gz -T | csvtk pretty -t --out-file seqkit_stats_v1.tsv

- [ ] It would be more helpful to provide as much information as you can:
    - [ ] Are you running on a personal computer or a server?
PC AMD Ryzen 9 5950X Processor, 32 threads, 128GB RAM, Ubuntu 20.04.6 LTS, Conda 22.9.0, loads of free RAM

    - [ ] Show the types and sizes of input files with `file xxx` and `ls -lh xxx`.
See file_stats.txt

    - [ ] Show some lines of input files with `head -n 5 xxx` or `zcat xxx.gz | head -n 5`.
See sequence_headers.txt

- [ ] Provide a reproducible example.
See below

    - [ ] Has this problem happened many times?
 Every parameter I have tried. See below.

    - [ ] Or it only failed with this input file or/and these command/parameters.
   Failed with:
Multithread option: -j 28 
Input file option: --infile-list files.txt
TSV out file option: -T | csvtk pretty -t --out-file seqkit_stats_v1.tsv

**Detailed problem:**
Whenever I run the "stats" option on my 488 FASTQ.gz files I get the following screen output:

(base) doudna@doudna:/mnt/linux_fast/bioinf_fast/GACU2024v1/top$ seqkit stats ./raw_unclean/lane_1/*.FASTQ.gz -T | csvtk pretty -t --out-file seqkit_stats_v1.tsv
processed files:  39 / 488 [==>-----------------------------------] ETA: 3m19s
processed files:  488 / 488 [======================================] ETA: 0s. done
(base) doudna@doudna:/mnt/linux_fast/bioinf_fast/GACU2024v1/top$ 

The process always processes 28 - 40 files before then "restarting" and processing the whole 488. However, I get no stats output unless I specify the --out-file option, in which case only the first 28 - 40 files have stats outputted but none of the others (see seqkit_stats_v1.tsv). The first number of files processed (see output) varies and sometimes in the output it will skip some .FASTQ.gz files and report subsequent ones but I think this may only happen with the multithread -j option.

What am I doing wrong? Your suite of tools are very impressive and I really want to use them but I can get past this first issue.

Thanks heaps for your help.

Cheers,
Dean

[file_stats.txt](https://github.com/user-attachments/files/15573438/file_stats.txt)
[sequence_headers.txt](https://github.com/user-attachments/files/15573439/sequence_headers.txt)
[seqkit_stats_v1.tsv.txt](https://github.com/user-attachments/files/15573457/seqkit_stats_v1.tsv.txt)

**Comment by shenwei356** [comment]
```
seqkit stats ./raw_unclean/lane_1/*.FASTQ.gz -T | csvtk pretty -t --out-file seqkit_stats_v1.tsv
```

I've used `seqkit stats` with 2 millions of files recently, and I didn't meet any problem. 

- For the "restart", the progress bar might not be well configured. It might clear some lines in terminal if you did not use `-o` or pipe to other commands, so you might think some lines are missing. The progress bar is outputed to stderr, which should not affect the output piped to `csvtk`. Are the 41 lines in `seqkit_stats_v1.tsv.txt` all what you got?
- `csvtk pretty`  read the first n (the default value might be 128 or 1000, depending on the version) rows to determin the minimum and maximum widths of each columns.

I suggest running the two commands one by one, not with pipe, to chech the possible problem.

```
seqkit stats -j 32 ./raw_unclean/lane_1/*.FASTQ.gz -T -o seqkit_stats_v1.tsv

# check the count
wc -l seqkit_stats_v1.tsv

# then view the result
csvtk pretty -t seqkit_stats_v1.tsv | more
```

**Comment by neutronstar21** [comment]
Thanks for your reply. I've tracked down the problem. I have 4 fastq.gz files that have malformed records.

Using bbtools and seqkit sana I was able to track them down:
[INFO][0m File: ./raw_unclean/lane1/2634618.fastq.gz	Pass records: 1720099	Discarded lines: 1528155
[INFO][0m File: ./raw_unclean/lane1/2634782.fastq.gz	Pass records: 1571417	Discarded lines: 1075311
[INFO][0m File: ./raw_unclean/lane1/2634796.fastq.gz	Pass records: 1925202	Discarded lines: 416341
[INFO][0m File: ./raw_unclean/lane1/2634912.fastq.gz	Pass records: 2559609	Discarded lines: 21891

For future releases, it would be great if the "stats" option ran "sana" when processing each file to produce an indication of unreadable files rather than "stats" failing silently for this error as it does now.

Anyhow, the problem has been identified, and I thank you for your time and this great suite of tools.

Cheers,
Dean
